src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageActionDescriptorBuilder.cs

```diff
@@ -40,6 +40,7 @@ public static class CompiledPageActionDescriptorBuilder
                 FilterDescriptors = filters,
                 HandlerMethods = handlerMethods,
                 HandlerTypeInfo = applicationModel.HandlerType,
+                DeclaredModelTypeInfo = applicationModel.HandlerType,
```

Comment:
This isn't always the case.  Pages can have a separate handler type and model type.

Reply:
Doesn't the `PageApplicationModel` have a corresponding property? Looks like you added it.

Reply:
Yeah, this is a bug, see above